### PR TITLE
[REF] AuthX - flatten already logged in cases

### PR DESCRIPTION
**Pull request by ufundo**
Overview
----------------------------------------
Attempt to flatten the existing code path logic, in order to make @aydun 's life easier trying to fix https://lab.civicrm.org/dev/core/-/issues/4463

Before
----------------------------------------
A locally defined function that includes checking a var is `!empty`, which is always passed values that are defined, inside a conditional where you already know one of them is truthy, inside a general conditional nest where it's hard to know whether you proceed or reject or complete.

After
----------------------------------------
-  helper function with specific cases for proceed and complete; falling back to reject if in doubt.
- Marginally more informative error message when rejecting

### CODE REVIEW

**Comment by totten** [comment]
Got my head in the other PR, but just a terminology note: this would be [REF not NFC](https://docs.civicrm.org/dev/en/latest/tools/git/#acronyms).

**Comment by eileenmcnaughton** [comment]
@aydun on a quick skim this seems probably mergeable?

**Comment by eileenmcnaughton** [comment]
OK - this is more readable - the other was hard to follow

File: ext/authx/Civi/Authx/Authenticator.php [hunk 1]
```diff
@@ -250,6 +250,36 @@ protected function checkPolicy(AuthenticatorTarget $tgt) {
...
+   * Check for a pre-existing login
+   *
+   * @param AuthenticatorTarget $tgt
+   * @return bool
+   */
+  protected function checkAlreadyLoggedIn(AuthenticatorTarget $tgt) {
+    $existingContact = \CRM_Core_Session::getLoggedInContactID();
+    $existingUser = $this->authxUf->getCurrentUserId();
+
+    if (!$existingContact && !$existingUser) {
```
Comment (eileenmcnaughton) [inline ext/authx/Civi/Authx/Authenticator.php:263]: If I put an OR here then then it works for me

Comment (eileenmcnaughton) [reply ext/authx/Civi/Authx/Authenticator.php:263]: ie what works is a situation where coworker is running intiates a login from NULL to something

**Comment by eileenmcnaughton** [comment]
So this is definitely more readable  & I agree the behaviour is unchanged
